sandbox/modules/identity/src/internalClusterTest/java/org/opensearch/identity/BasicAuthenticationIT.java

```diff
@@ -0,0 +1,186 @@
+                                    interceptedTokens.put(
+                                        request.getParentTask().getNodeId(),
+                                        tcHeaders.get(ThreadContextConstants.OPENSEARCH_AUTHENTICATION_TOKEN_HEADER)
+                                    );
+                                }
+                            }
+                            // String prefix = "(nodeName=" + request.getParentTask().getNodeId() + ", requestId=" +
+                            // request.getParentTask().getId() + ", action=" + action + " interceptSender)";
+                            // System.out.println(prefix + " Headers: " + threadContext.getHeaders());
+                            sender.sendRequest(connection, action, request, options, handler);
+                        }
+                    };
+                }
+            });
+        }
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        List<Class<? extends Plugin>> plugins = super.nodePlugins().stream().collect(Collectors.toList());
+        plugins.add(TokenInterceptorPlugin.class);
+        return plugins;
+    }
+
+    public void testBasicAuth() throws Exception {
+        final String clusterManagerNode = internalCluster().startClusterManagerOnlyNode();
+
+        ClusterStateResponse clusterStateResponse = client(clusterManagerNode).admin()
+            .cluster()
+            .prepareState()
+            .setClusterManagerNodeTimeout("1s")
+            .clear()
+            .setNodes(true)
+            .get();
+        assertNotNull(clusterStateResponse.getState().nodes().getClusterManagerNodeId());
+
+        // start another node
+        final String dataNode = internalCluster().startDataOnlyNode();
+        clusterStateResponse = client(dataNode).admin()
+            .cluster()
+            .prepareState()
+            .setClusterManagerNodeTimeout("1s")
+            .clear()
+            .setNodes(true)
+            .setLocal(true)
+            .get();
+        assertNotNull(clusterStateResponse.getState().nodes().getClusterManagerNodeId());
+        // wait for the cluster to form
+        assertNoTimeout(client().admin().cluster().prepareHealth().setWaitForNodes(Integer.toString(2)).get());
+        List<NodeInfo> nodeInfos = client().admin().cluster().prepareNodesInfo().get().getNodes();
+        assertEquals(2, nodeInfos.size());
+
+        List<TransportService> transportServices = new ArrayList<TransportService>();
+        Map<String, TransportMessageListener> listenerMap = new HashMap<>();
+
+        TransportService clusterManagerService = internalCluster().getInstance(TransportService.class, clusterManagerNode);
+        transportServices.add(clusterManagerService);
+
+        TransportService dataNodeService = internalCluster().getInstance(TransportService.class, dataNode);
+        transportServices.add(dataNodeService);
+
+        for (TransportService service : transportServices) {
```

Comment:
Cool this works, maybe there is an easier way to set this up, or we should abstract it.  I think there will be many more scenarios in spaces like this, what do you think?

Reply:
Yes, I was really wrestling with this test trying to trace the authc information throughout the cluster. In this case the datanode forwards the transport request to the cluster manager and it asserts that authc information is received at the cluster manager and sent from the data node.